modules/AsyncProps.js

```diff
@@ -245,17 +206,24 @@ class AsyncProps extends React.Component {
     if (!routeChanged)
       return
 
-    const oldComponents = filterAndFlattenComponents(this.props.components)
-    const newComponents = filterAndFlattenComponents(nextProps.components)
-    let components = arrayDiff(oldComponents, newComponents)
-
-    if (components.length === 0) {
-      const sameComponents = shallowEqual(oldComponents, newComponents)
-      if (sameComponents) {
-        const paramsChanged = !shallowEqual(nextProps.params, this.props.params)
-        if (paramsChanged)
-          components = [ last(newComponents) ]
+    let components = []
+    let pivoted = false
+    for (let i = 0; i < nextProps.routes.length; i++) {
```

Comment:
why iterate routes when the router already gives us the components to be rendered?